src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/ReadOnlyTagHelperAttributeList.cs

```diff
@@ -166,41 +165,49 @@ public int IndexOf(TAttribute item)
         }
 
         /// <summary>
-        /// Retrieves the first <typeparamref name="TAttribute"/> with <see cref="IReadOnlyTagHelperAttribute.Name"/>
+        /// Retrieves the first <typeparamref name="TagHelperAttribute"/> with <see cref="TagHelperAttribute.Name"/>
         /// matching <paramref name="name"/>.
         /// </summary>
-        /// <param name="name">The <see cref="IReadOnlyTagHelperAttribute.Name"/> of the
-        /// <typeparamref name="TAttribute"/> to get.</param>
-        /// <param name="attribute">When this method returns, the first <typeparamref name="TAttribute"/> with
-        /// <see cref="IReadOnlyTagHelperAttribute.Name"/> matching <paramref name="name"/>, if found; otherwise,
+        /// <param name="name">The <see cref="TagHelperAttribute.Name"/> of the
+        /// <typeparamref name="TagHelperAttribute"/> to get.</param>
+        /// <param name="attribute">When this method returns, the first <typeparamref name="TagHelperAttribute"/> with
+        /// <see cref="TagHelperAttribute.Name"/> matching <paramref name="name"/>, if found; otherwise,
         /// <c>null</c>.</param>
-        /// <returns><c>true</c> if a <typeparamref name="TAttribute"/> with the same
-        /// <see cref="IReadOnlyTagHelperAttribute.Name"/> exists in the collection; otherwise, <c>false</c>.</returns>
+        /// <returns><c>true</c> if a <typeparamref name="TagHelperAttribute"/> with the same
+        /// <see cref="TagHelperAttribute.Name"/> exists in the collection; otherwise, <c>false</c>.</returns>
         /// <remarks><paramref name="name"/> is compared case-insensitively.</remarks>
-        public bool TryGetAttribute(string name, out TAttribute attribute)
+        public bool TryGetAttribute(string name, out TagHelperAttribute attribute)
         {
             if (name == null)
             {
                 throw new ArgumentNullException(nameof(name));
             }
 
-            attribute = this[name];
+            attribute = null;
+            for (var i = 0; i < Attributes.Count; i++)
+            {
+                if (NameEquals(name, Attributes[i]))
+                {
+                    attribute = Attributes[i];
+                    break;
+                }
+            }
```

Comment:
Undo this change once https://github.com/aspnet/Razor/pull/682#discussion-diff-51338876 is fixed.